### PR TITLE
Remove styled-jsx in favor of Styled Components

### PR DESCRIPTION
**Pull request by SudharakaP**
This removes styles-jsx in favor of styled components.

Part of https://github.com/opencollective/opencollective/issues/3289

### CODE REVIEW

File: components/InputTypeLocation.js [hunk 1]
```diff
@@ -4,10 +4,82 @@ import classNames from 'classnames';
...
+  }
+  .geosuggest__suggests--hidden {
+    max-height: 0;
+    overflow: hidden;
+    border-width: 0;
+  }
+
+  /**
+* A geosuggest item
+*/
```
Comment (Betree) [inline components/InputTypeLocation.js:62]:
```suggestion
  * A geosuggest item
  */
```

File: components/InputTypeLocation.js [hunk 2]
```diff
@@ -65,78 +137,6 @@ class InputTypeLocation extends React.Component {
 
     return (
       <div className={classNames('InputTypeLocation', this.props.className)}>
```
Comment (Betree) [inline components/InputTypeLocation.js:139]: I don't think we need this anymore

```suggestion
      <div>
```

**Review by Betree — CHANGES_REQUESTED** [review]
There's a visual regression on `InputTypeLocation`, the list doesn't appear:
image

Instead it appears at the bottom of the page:
image

**Comment by SudharakaP** [comment]
@Betree : Thanks. Let me correct those issue along with the other migrations which I am doing right now. 😄

File: components/Body.js [hunk 3]
```diff
@@ -27,107 +121,7 @@ export default class Body extends React.Component {
...
-            @media (max-width: 420px) {
-              .content {
-                padding: 1rem 1.5rem;
-              }
-            }
-          `}
-        </style>
-        {this.props.children}
-      </Main>
+      <Main withBorderTop={withBorderTop}>{this.props.children}</Main>
```
Comment (Betree) [inline components/Body.js:124]: We should not include global styles if `withoutGlobalStyles` is false.

File: components/CollectivesWithData.js [hunk 4]
```diff
@@ -4,12 +4,42 @@ import { gql } from '@apollo/client';
...
 import { FormattedMessage } from 'react-intl';
+import styled from 'styled-components';
 
 import CollectiveCard from './CollectiveCard';
 import Error from './Error';
 
 const COLLECTIVE_CARDS_PER_PAGE = 10;
 
+const CollectivesContainer = styled.div`
+  :global(.loadMoreBtn) {
```
Comment (Betree) [inline components/CollectivesWithData.js:15]: I don't think `:global` works with styled components. Does it?

**Review by Betree — CHANGES_REQUESTED** [review]
Thanks @SudharakaP, this represent some work but there's a major issue with most of the components migrated: using classes is discouraged. With styled-components, the goal is to apply styles at the component level. So using classes/writing this is problematic:

```jsx
styled.div`
  .myComponent { ... }
`
```

We should instead style dedicated components, or use the existing StyledComponents (`Box`, `Container`, `Flex`, etc).

The goal is to get rid of styled-jsx, but we don't want to bring the debt into styled-components. The `InputTypeLocation` is an exception to that, because we don't have control over the DOM it generates.

For this reason we can't go with the changes in this PR. To go further with this issue I suggest opening one PR per migrated component, it will ensure that changes are easier to test, to review and to rollback if needed.

**Comment by SudharakaP** [comment]
> Thanks @SudharakaP, this represent some work but there's a major issue with most of the components migrated: using classes is discouraged. With styled-components, the goal is to apply styles at the component level. So using classes/writing this is problematic:
> 
> ```js-jsx
> styled.div`
>   .myComponent { ... }
> `
> ```
> 
> We should instead style dedicated components, or use the existing StyledComponents (`Box`, `Container`, `Flex`, etc).
> 
> The goal is to get rid of styled-jsx, but we don't want to bring the debt into styled-components. The `InputTypeLocation` is an exception to that, because we don't have control over the DOM it generates.
> 
> For this reason we can't go with the changes in this PR. To go further with this issue I suggest opening one PR per migrated component, it will ensure that changes are easier to test, to review and to rollback if needed.

@Betree : Ah, got it. Let me close this PR and start from the beginning. I'll open separate PRs. 😄

File: components/Body.js (hunk 3 above)
Comment (SudharakaP) [reply components/Body.js:124]: Will keep in mind when I open the new PRs. Thanks. 👍🏼

File: components/CollectivesWithData.js (hunk 4 above)
Comment (SudharakaP) [reply components/CollectivesWithData.js:15]: Let me double check when I open new PRs. But seemed like it was working. 👍🏼

**Pull request closed without merging**